app/modules/individuals/models.py

```diff
@@ -229,14 +229,59 @@ def merge(cls, *individuals, sex=None, primary_name=None):
                     log, target_individual, f'assigned {enc} from merged {indiv}'
                 )
                 enc.individual_guid = target_individual.guid
-            # TODO also consolidate SocialGroups
+            target_individual._consolidate_social_groups(indiv)
             indiv.delete()
         return result
 
     def merge_from(self, *individuals):
         all_indiv = (self,) + individuals
         return Individual.merge(*all_indiv)
 
+    # mimics Individual.merge() in args, but does not immediately executes; rather waits for approval
+    #   and initiates a request including time-out etc
+    @classmethod
+    def merge_request(cls, *individuals, sex=None, primary_name=None):
+        if not individuals or not isinstance(individuals, tuple) or len(individuals) < 2:
+            raise ValueError('must be passed a tuple of at least 2 individuals')
+        target_individual = individuals[0]
+        source_individuals = individuals[1:]
+        log.warning(
+            f'merge_request() on {target_individual} from {source_individuals} -- NOT YET IMPLEMENTED'
+        )
+        return False
+
+    def merge_request_from(self, *individuals):
+        all_indiv = (self,) + individuals
+        return Individual.merge_request(*all_indiv)
+
+    def _consolidate_social_groups(self, source_individual):
+        if not source_individual.social_groups:
+            return
+        for source_member in source_individual.social_groups:
+            socgrp = source_member.social_group
+            already_member = socgrp.get_member(str(self.guid))
+            data = {'roles': source_member.roles}  # roles may be empty, this is fine
+            # must blow away source_member roles in case one is singular (as it will get passed to target)
+            source_member.roles = None
+            with db.session.begin(subtransactions=True):
+                db.session.merge(source_member)
+            if already_member:
```

Comment:
This definitely should be a method on the group. SocialGroup.add_roles passing in the new roles. 
The changing of data within a module should always be done within the models class for that module.

Reply:
`add_roles(self, individual_guid, roles)` added